libyul/backends/evm/EVMObjectCompiler.cpp

```diff
@@ -77,7 +77,7 @@ void EVMObjectCompiler::run(Object const& _object, bool _optimize)
 	yulAssert(_object.hasCode(), "No code.");
 	if (evmDialect->eofVersion().has_value())
 		yulAssert(
-			_optimize && (evmDialect->evmVersion() >= langutil::EVMVersion::prague()),
+			_optimize && evmDialect->evmVersion().supportsEOF(),
 			"Experimental EOF support is only available for optimized via-IR compilation and the most recent EVM version."
 		);
```

Comment:
If we're touching it, I'd correct a two minor things here:
- I don't think we have a validation against `--optimize` in the CLI or Standard JSON interface. So this can't be a plain assert. Please change it `solUnimplementedAssert()`, which will generate a proper error.
    - On the other hand, the EVM version check is correct as an assert.
- I'd drop any mentions of "experimental" from those messages since it it's implied in the option name anyway and we're very likely to forget to remove them when we make it non-experimental.

```suggestion
		solUnimplementedAssert(_optimize, "EOF supported only for optimized compilation via IR.");
		yulAssert(evmDialect->evmVersion().supportsEOF());
```